packages/app/src/cli/models/app/loader.ts

```diff
@@ -414,7 +422,51 @@ class AppLoader {
       }
     })
 
-    return {allExtensions, usedCustomLayout: extensionDirectories !== undefined}
+    // Load app config modules
+    const appConfigModules = await this.loadAppConfigModules(appDirectory)
+
+    return {allExtensions: allExtensions.concat(appConfigModules), usedCustomLayout: extensionDirectories !== undefined}
+  }
+
+  async loadAppConfigModules(directory: string) {
+    const {configuration} = await loadAppConfiguration({
+      configName: undefined,
+      directory,
+    })
+
+    const appConfigSpecifications = this.specifications.filter((specification) =>
+      specification.appModuleFeatures().includes('app_config'),
+    )
+
+    const appConfigModules: ExtensionInstance[] = []
+
+    await Promise.all(
+      appConfigSpecifications
+        .map(async (specification) => {
+          // Skip creating an app access module unless `access` is configured
+          if (
+            specification.identifier === APP_ACCESS_IDENTIFIER &&
+            !(configuration as CurrentAppConfiguration).access
+          ) {
+            return
+          }
+
+          const promise = this.createExtensionInstance(
+            specification.identifier,
+            configuration,
+            configuration.path,
+            directory,
+          ).then((module) => {
+            if (module) {
+              appConfigModules.push(module)
+            }
+          })
+          return promise
```

Comment:
I think we can get away with only using `async/await` without mixing in promise chaining too. Have not tested this and might need a tweak, but the idea is there

```ts
const validAppConfigSpecifications = appConfigSpecifications.filter(specification => specification.identifier !== APP_ACCESS_IDENTIFIER &&
            (configuration as CurrentAppConfiguration).access)
// that ^ will probably have to change to support the other multiple types of app config specifications down the line too

const appConfigModulePromises = validAppConfigSpecifications.map(specification => this.createExtensionInstance(
            specification.identifier,
            configuration,
            configuration.path,
            directory,
          )).filter(Boolean) // for the if(module) check, this can possibly not return a module? also wasnt sure where the .flat() was needed, may need to tweak that

return Promise.all(appConfigModulePromises)
```

Reply:
What do you think of a refactor like this?

```ts
  async loadAppConfigModules(directory: string) {
    const {configuration: config} = await loadAppConfiguration({configName: undefined, directory})
    const {path} = config
    const {createExtensionInstance, specifications} = this

    const filteredSpecs = specifications.filter(({appModuleFeatures, identifier}) => {
      const hasAppConfigFeature = appModuleFeatures().includes('app_config')
      const hasAppAccessIdentifier = identifier === APP_ACCESS_IDENTIFIER
      const hasAccessConfig = (config as CurrentAppConfiguration).access
      return hasAppConfigFeature && hasAppAccessIdentifier && hasAccessConfig
    })

    const promises = filteredSpecs.map(({identifier}) => createExtensionInstance(identifier, config, path, directory))
    const filteredPromises = promises.filter(Boolean)
    const result = await Promise.all(filteredPromises)
    return result
  }
```

>also wasnt sure where the .flat() was needed, may need to tweak that

🤔 I'm also not sure why the `.flat()` was there. That was there when I took over the PR.

Reply:
I like it! Only thing is an `async` function will always return a promise, so you don't have to await the data before returning since you'll do that any way on the other side (unless we don't :thonk, but that would be bad if we did)

```js
         return Promise.all(filteredPromises)
```